plugins/woocommerce/src/Utilities/NumberUtil.php

```diff
@@ -49,4 +49,63 @@ public static function array_sum( array $arr ): float {
 
 		return array_sum( $sanitized_array );
 	}
+
+	/**
+	 * Accepts a cost using the current locale and returns it expressed as a numeric string.
+	 *
+	 * @param string $value The cost to sanitize, containing only numbers and the current locale's separators.
+	 * @return string The sanitized cost as a numeric string (passing is_numeric()).
+	 * @throws \InvalidArgumentException If the value contains invalid characters (including "." or " " if they are not the current locale's separators).
+	 */
+	public static function sanitize_cost_in_current_locale( $value ): string {
+		$value = is_null( $value ) ? '' : $value;
+		$value = wp_kses_post( trim( wp_unslash( $value ) ) );
+		$currency_symbol_encoded = get_woocommerce_currency_symbol();
+		$currency_symbol_variations = array( $currency_symbol_encoded, wp_kses_normalize_entities( $currency_symbol_encoded ), html_entity_decode( $currency_symbol_encoded ) );
+		
+		$value = str_replace( $currency_symbol_variations, '', $value );
+
+		$allowed_characters_regex = sprintf(
+			'/^[0-9\%s\%s]*$/',
+			wc_get_price_thousand_separator(),
+			wc_get_price_decimal_separator()
+		);
+
+		if ( 1 !== preg_match( $allowed_characters_regex, $value ) ) {
+			throw new \InvalidArgumentException( sprintf( __( '%s is not a valid numeric value. Allowed characters: %s', 'woocommerce' ), $value, $allowed_characters_regex ) );
+		}
```

Comment:
This is great feedback when things go wrong! We should offer a more merchant friendly message though. How about something like: `1oooo is not a valid numeric value. Please make sure to enter a valid number`

<img width="648" alt="image" src="https://github.com/user-attachments/assets/4c33a79f-1682-40bf-bf59-07a734123e4b" />

Reply:
I had something like that initially but it just seemed to previous less information and the second sentence doesn't help the user troubleshoot 🙈 I could translate the regex to english?

Reply:
Not sure if that's helpful but we could also include the separators we're talking about in the message, like:

> [...] are the thousand (`,`) and decimal (`.`) separators.